sound/soc/sof/trace.c

```diff
@@ -151,6 +151,9 @@ int snd_sof_init_trace_ipc(struct snd_sof_dev *sdev)
 	struct sof_ipc_reply ipc_reply;
 	int ret;
 
+	if (sdev->dtrace_is_enabled)
+		return;
```

Comment:
sound/soc/sof/trace.c: In function ‘snd_sof_init_trace_ipc’:
sound/soc/sof/trace.c:155:3: warning: ‘return’ with no value, in function returning non-void [-Wreturn-type]
   return;
   ^~~~~~
sound/soc/sof/trace.c:148:5: note: declared here
 int snd_sof_init_trace_ipc(struct snd_sof_dev *sdev)